src/pages/test-result/test-result.ts

```diff
@@ -39,4 +70,154 @@ export class TestResultPage {
   goBack() {
     this.navCtrl.pop({ animate: false });
   }
+
+  toggleStopOrDestroy = () => {
+    if (this.stopOrDestroyRecording === 'stop') {
+      this.stopRecordingAudio();
+    }
+    if (this.stopOrDestroyRecording === 'destroy') {
+      this.deleteAudio();
+    }
+    this.stopOrDestroyRecording = this.stopOrDestroyRecording === 'stop' ? 'destroy' : 'stop';
+  };
+
+  togglePlayOrPause = () => {
+    if (this.playOrPauseRecording === 'play') {
+      if (this.isRecording) {
+        this.resumeRecording();
+      } else {
+        this.playAudio();
+      }
+    }
+    if (this.playOrPauseRecording === 'pause') {
+      if (this.isRecording) {
+        this.pauseRecording();
+      } else {
+        this.pauseAudio();
+      }
+    }
+    this.playOrPauseRecording = this.playOrPauseRecording === 'play' ? 'pause' : 'play';
+  };
+
+  // audio debrief
+  recordAudio() {
+    if (!this.isRecording) {
+      this.file
+        .createFile(this.file.dataDirectory, this.fileName, true)
+        .then(() => {
+          const src = this.file.dataDirectory.replace(/^file:\/\//, '') + this.fileName;
+          // ionic media.create is buggy (getDuration returns always -1), the solution is to use raw cordova API
+          this.audio = new (<any>window).Media(
+            src,
+            this.mediaSuccess,
+            this.mediaError,
+            this.mediaStatus
+          );
+          this.audio.startRecord();
+          this.isRecording = true;
+          this.stopOrDesDisabled = false;
+          this.playOrPauseDisabled = false;
+          this.playOrPauseRecording = 'pause';
+        })
+        .catch(this.logError);
+    }
+  }
+
+  stopRecordingAudio() {
+    console.log('Stopping recording');
+
+    const self = this;
```

Comment:
We shouldn't need to do this anymore, when we use => for functions, the this scope makes sense